### PR TITLE
JAPI-248 Provide crypto support

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds commons/utils CryptoHelper
- Adds set/fetch encrypted wsstore calls
- Adds junit test cases

Signed-off-by: Pastrana, Rodrigo <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@drealeed please take a look and let me know if this is inline with the work we discussed
@jpmcmu please review

File: commons-hpcc/src/main/java/org/hpccsystems/commons/utils/CryptoHelper.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        try
+        {
+            if (cipher != null)
+            {
+                return Base64.getEncoder().encodeToString(cipher.doFinal(utf8StrToEncrypt.getBytes("UTF-8")));
+            }
+            else
+            {
+                throw new Exception("Must provide chiper");
+            }
```
Comment (drealeed) [inline commons-hpcc/src/main/java/org/hpccsystems/commons/utils/CryptoHelper.java]: spelling - cipher

**Comment by rpastrana** [comment]
Fixed @drealeed 
@jpmcmu please review

File: commons-hpcc/src/main/java/org/hpccsystems/commons/utils/CryptoHelper.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+import javax.crypto.Cipher;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.log4j.Logger;
+
+public class CryptoHelper
+{
+    private final static Logger log = Logger.getLogger(CryptoHelper.class.getName());
+
+    public final static String DEFAULT_DIGEST_ALGORITHM = "SHA-512";
```
Comment (jpmcmu) [inline commons-hpcc/src/main/java/org/hpccsystems/commons/utils/CryptoHelper.java]: @rpastrana Could we make the algorithm options an enum?

File: commons-hpcc/src/main/java/org/hpccsystems/commons/utils/CryptoHelper.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+            System.out.println("Encrypted using custom cipher: " + encryptedString);
+            System.out.println("Decrypted string custom cipher: " + decryptedString);
+            System.out.println("Encrypted using default cipher: " + encryptedDefaultCipherString);
+            System.out.println("Decrypted using default cipher: " + decryptedDefaultCipherString);
+        }
+        catch (Exception e)
+        {
+            log.error("Could not create cipher");
+            e.printStackTrace();
+        }
```
Comment (jpmcmu) [inline commons-hpcc/src/main/java/org/hpccsystems/commons/utils/CryptoHelper.java]: Should the be part of a test?

**Comment by rpastrana** [comment]
@jpmcmu please review again.

**Pull request merged**